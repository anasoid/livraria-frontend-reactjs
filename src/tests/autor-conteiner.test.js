import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import AutorConteiner from '../components/autor/autor-conteiner';

describe('test AutorConteiner component', () => {

  const wrapper = shallow(<AutorConteiner/>);

  it('should renders without crashing', () => {
    shallow(<AutorConteiner/>);
    const tree = renderer.create(<AutorConteiner />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('AutorConteiner should be contain form', () => {
    expect(wrapper.find('form')).toExist();
  });

  it('should be contains a "Novo Autor" title', () => {
    expect(wrapper.find('h1')).toHaveText('Novo Autor');
  });

  it('should be contains nome input', () => {
    expect(wrapper.find('#autorNome')).toMatchSelector('input');
  });

  it('should be contains email input', () => {
    expect(wrapper.find('#autorEmail')).toMatchSelector('input');
  });

  it('should be contain a button "Gravar Autor" and type submit', () => {
    const button = wrapper.find('#btnGravarAutor');
    expect(button).toHaveProp('type', 'submit');
    expect(button).toHaveText('Gravar Autor');
  });

  it('should be contain a table with title "Autores"', () => {
    const table = wrapper.find('table');
    expect(table.find('caption')).toHaveText('Autores');
  });

  it('should be contain a table with columns Nome, Email', () => {
    const rowsTh = wrapper.find('table').find('thead').find('tr').find('th');
    expect(rowsTh.at(0)).toHaveText('Nome');
    expect(rowsTh.at(1)).toHaveText('Email');
  });

});