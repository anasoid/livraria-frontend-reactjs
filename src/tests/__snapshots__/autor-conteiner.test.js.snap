// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AutorConteiner component should renders without crashing 1`] = `
<div>
  <h1>
    Novo Autor
  </h1>
  <form>
    <label
      htmlFor="autorNome"
    >
      Nome: 
    </label>
    <input
      id="autorNome"
    />
    <br />
    <label
      htmlFor="autorEmail"
    >
      Email: 
    </label>
    <input
      id="autorEmail"
    />
    <br />
    <button
      id="btnGravarAutor"
      type="submit"
    >
      Gravar Autor
    </button>
  </form>
  <br />
  <table>
    <caption>
      Autores
    </caption>
    <thead>
      <tr>
        <th>
          Nome
        </th>
        <th>
          Email
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Fulano
        </td>
        <td>
          fulano@admin.com
        </td>
        <td>
          <button>
            alterar
          </button>
        </td>
        <td>
          <button>
            remover
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
